File: DAA/Lab 06/02_question2.py
def find_min_max(arr, left, right):
    if left == right:
        return arr[left], arr[right]
    
    if right == left + 1:
        return (min(arr[left], arr[right]), max(arr[left], arr[right]))
    
    mid = (left + right) // 2
    left_min, left_max = find_min_max(arr, left, mid)
    right_min, right_max = find_min_max(arr, mid + 1, right)
    
    return (min(left_min, right_min), max(left_max, right_max))

arr = [3, 5, 1, 8, 2, 7, 6, 4]
min_val, max_val = find_min_max(arr, 0, len(arr) - 1)
print(f"Minimum: {min_val}, Maximum: {max_val}")

File: DAA/Lab 07/02_question2.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

pair<int, int> min_max(vector<int>& v, int s, int e) {
    if (s == e) return {v[s], v[s]};

    int mid = (s + e) / 2;
    pair<int, int> left = min_max(v, s, mid);
    pair<int, int> right = min_max(v, mid + 1, e);

    int minval = min(left.first, right.first);
    int maxval = max(left.second, right.second);

    return {minval, maxval};
}

int main() {
    int n;
    cin >> n;
    vector<int> v(n);
    for (int i = 0; i < n; i++) cin >> v[i];

    pair<int, int> result = min_max(v, 0, n - 1);
    cout << "Max Distance: " << result.second - result.first << endl;
}